tests/helpers/test_entity.py

```diff
@@ -1911,6 +1911,64 @@ async def async_will_remove_from_hass(self):
     assert ent._platform_state == entity.EntityPlatformState.ADDED
 
 
+@pytest.mark.parametrize("config_subentry_id", [None, "mock-subentry-id-1"])
+async def test_change_entity_id_entry_integrity(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    config_subentry_id: str | None,
+) -> None:
+    """Test changing entity id sets the subentry id correctly."""
+
+    class MockEntity(entity.Entity):
+        _attr_unique_id = "5678"
+
+    async def async_setup_entry(
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        async_add_entities: AddConfigEntryEntitiesCallback,
+    ) -> None:
+        """Mock setup entry method."""
+        async_add_entities([MockEntity()], config_subentry_id=config_subentry_id)
+
+    platform = MockPlatform(async_setup_entry=async_setup_entry)
+    config_entry = MockConfigEntry(
+        entry_id="super-mock-id",
+        subentries_data=[
+            ConfigSubentryData(
+                data={},
+                subentry_id="mock-subentry-id-1",
+                subentry_type="test",
+                title="Mock title",
+                unique_id="test",
+            ),
+        ],
+    )
+    config_entry.add_to_hass(hass)
+    entity_platform = MockEntityPlatform(
+        hass, platform_name=config_entry.domain, platform=platform
+    )
+
+    assert await entity_platform.async_setup_entry(config_entry)
+    await hass.async_block_till_done()
+
+    ent = entity_registry.async_get(next(iter(hass.states.async_entity_ids())))
+    assert ent is not None
+
+    state = hass.states.async_all()[0]
+    assert state is not None
```

Comment:
Instead of just asserting not `None`, why not assert an entity id and compare the registry entry to a snapshot?